BUG: stats: XSLOW test failures in `test_continuous_basic`

### Describe your issue.

With the environment variable `SCIPY_XSLOW=1` set, I am getting 19 test failures from the `test_continous_basic::test_cont_basic_fit` test suite in `stats`.  All the tests hit the same bug, so I cut out all but the first two detailed reports in the following transcript.

<details>
<summary>Test run</summary>

```
$ SCIPY_XSLOW=1 python3 dev.py test -m full -t scipy.stats.tests.test_continuous_basic::test_cont_basic_fit -j 6
💻  ninja -C /home/warren/repos/git/forks/scipy/build -j12
ninja: Entering directory `/home/warren/repos/git/forks/scipy/build'
[2/2] Generating scipy/generate-version with a custom command
Build OK
💻  meson install -C build --only-changed
Installing, see meson-install.log...
Installation OK
SciPy from development installed path at: /home/warren/repos/git/forks/scipy/build-install/lib/python3.12/site-packages
Running tests for scipy version:1.14.0.dev0+1021.bd60b4e, installed at:/home/warren/repos/git/forks/scipy/build-install/lib/python3.12/site-packages/scipy
============================= test session starts ==============================
platform linux -- Python 3.12.3, pytest-8.1.1, pluggy-1.5.0
rootdir: /home/warren/repos/git/forks/scipy
configfile: pytest.ini
plugins: timeout-2.3.1, xdist-3.5.0, hypothesis-6.100.1, cov-5.0.0
6 workers [372 items]   
s....s.s.......s...................................s..s..........s...... [ 19%]
s..s.......................s....s....s.............sF..sFF.............s [ 38%]
..sF..........s..........s......s.........s..sF.........s............... [ 58%]
.F..F........F......F..........s....F..ss.......F..........ss......s.... [ 77%]
.........................................FF..s.................F........ [ 96%]
..FFFFF...s.                                                             [100%]
=================================== FAILURES ===================================
_______________ test_cont_basic_fit[200-kappa4-arg179-None-None] _______________
[gw1] linux -- Python 3.12.3 /home/warren/py3.12.3/bin/python3
scipy/stats/tests/test_continuous_basic.py:241: in test_cont_basic_fit
    check_fit_args(distfn, arg, rvs, method)
        arg        = (0.1, 0.0)
        distfn     = <scipy.stats._continuous_distns.kappa4_gen object at 0x7a8096371c10>
        distname   = 'kappa4'
        fix_args   = None
        method     = None
        n_fit_samples = 200
        rng        = RandomState(MT19937) at 0x7A8070196040
        rvs        = array([-0.92611927, -0.39031297, -0.51810899,  2.74702685, -0.26140543,
        0.81347898,  1.53637757,  0.92829259, ...577439,  1.90613802,  2.13208781,  0.24871165,
        0.17240924, -0.81257477,  0.27244402,  0.04268143,  1.89862333])
scipy/stats/tests/test_continuous_basic.py:746: in check_fit_args
    vals = distfn.fit(rvs, method=method)
        arg        = (0.1, 0.0)
        distfn     = <scipy.stats._continuous_distns.kappa4_gen object at 0x7a8096371c10>
        method     = None
        rvs        = array([-0.92611927, -0.39031297, -0.51810899,  2.74702685, -0.26140543,
        0.81347898,  1.53637757,  0.92829259, ...577439,  1.90613802,  2.13208781,  0.24871165,
        0.17240924, -0.81257477,  0.27244402,  0.04268143,  1.89862333])
        sup        = <numpy.testing._private.utils.suppress_warnings object at 0x7a80643bdeb0>
scipy/stats/_distn_infrastructure.py:2635: in fit
    method = kwds.get('method', "mle").lower()
E   AttributeError: 'NoneType' object has no attribute 'lower'
        args       = ()
        data       = array([-0.92611927, -0.39031297, -0.51810899,  2.74702685, -0.26140543,
        0.81347898,  1.53637757,  0.92829259, ...577439,  1.90613802,  2.13208781,  0.24871165,
        0.17240924, -0.81257477,  0.27244402,  0.04268143,  1.89862333])
        kwds       = {'method': None}
        self       = <scipy.stats._continuous_distns.kappa4_gen object at 0x7a8096371c10>
_______________ test_cont_basic_fit[200-ksone-arg183-None-None] ________________
[gw1] linux -- Python 3.12.3 /home/warren/py3.12.3/bin/python3
scipy/stats/tests/test_continuous_basic.py:241: in test_cont_basic_fit
    check_fit_args(distfn, arg, rvs, method)
        arg        = (1000,)
        distfn     = <scipy.stats._continuous_distns.ksone_gen object at 0x7a8098cf84d0>
        distname   = 'ksone'
        fix_args   = None
        method     = None
        n_fit_samples = 200
        rng        = RandomState(MT19937) at 0x7A8070196D40
        rvs        = array([0.00512931, 0.01046228, 0.0091458 , 0.03708666, 0.01179505,
       0.02229693, 0.02840279, 0.02331973, 0.029174...11, 0.02285444, 0.03123931, 0.03288781, 0.01696762,
       0.01621168, 0.00619958, 0.01720114, 0.01490967, 0.03118342])
scipy/stats/tests/test_continuous_basic.py:746: in check_fit_args
    vals = distfn.fit(rvs, method=method)
        arg        = (1000,)
        distfn     = <scipy.stats._continuous_distns.ksone_gen object at 0x7a8098cf84d0>
        method     = None
        rvs        = array([0.00512931, 0.01046228, 0.0091458 , 0.03708666, 0.01179505,
       0.02229693, 0.02840279, 0.02331973, 0.029174...11, 0.02285444, 0.03123931, 0.03288781, 0.01696762,
       0.01621168, 0.00619958, 0.01720114, 0.01490967, 0.03118342])
        sup        = <numpy.testing._private.utils.suppress_warnings object at 0x7a80643bfb90>
scipy/stats/_distn_infrastructure.py:2635: in fit
    method = kwds.get('method', "mle").lower()
E   AttributeError: 'NoneType' object has no attribute 'lower'
        args       = ()
        data       = array([0.00512931, 0.01046228, 0.0091458 , 0.03708666, 0.01179505,
       0.02229693, 0.02840279, 0.02331973, 0.029174...11, 0.02285444, 0.03123931, 0.03288781, 0.01696762,
       0.01621168, 0.00619958, 0.01720114, 0.01490967, 0.03118342])
        kwds       = {'method': None}
        self       = <scipy.stats._continuous_distns.ksone_gen object at 0x7a8098cf84d0>

[...]

=========================== short test summary info ============================
FAILED scipy/stats/tests/test_continuous_basic.py::test_cont_basic_fit[200-kappa4-arg179-None-None] - AttributeError: 'NoneType' object has no attribute 'lower'
FAILED scipy/stats/tests/test_continuous_basic.py::test_cont_basic_fit[200-ksone-arg183-None-None] - AttributeError: 'NoneType' object has no attribute 'lower'
FAILED scipy/stats/tests/test_continuous_basic.py::test_cont_basic_fit[200-kstwo-arg184-None-None] - AttributeError: 'NoneType' object has no attribute 'lower'
FAILED scipy/stats/tests/test_continuous_basic.py::test_cont_basic_fit[200-levy_stable-arg202-None-None] - AttributeError: 'NoneType' object has no attribute 'lower'
FAILED scipy/stats/tests/test_continuous_basic.py::test_cont_basic_fit[200-ncx2-arg245-None-None] - AttributeError: 'NoneType' object has no attribute 'lower'
FAILED scipy/stats/tests/test_continuous_basic.py::test_cont_basic_fit[200-powerlognorm-arg272-None-None] - AttributeError: 'NoneType' object has no attribute 'lower'
FAILED scipy/stats/tests/test_continuous_basic.py::test_cont_basic_fit[200-powernorm-arg275-MM-None] - scipy.integrate._quadpack_py.IntegrationWarning: The integral is probably d...
FAILED scipy/stats/tests/test_continuous_basic.py::test_cont_basic_fit[200-recipinvgauss-arg284-None-None] - AttributeError: 'NoneType' object has no attribute 'lower'
FAILED scipy/stats/tests/test_continuous_basic.py::test_cont_basic_fit[200-rel_breitwigner-arg291-MM-None] - scipy.stats._warnings_errors.FitError: Optimization failed: either a data m...
FAILED scipy/stats/tests/test_continuous_basic.py::test_cont_basic_fit[200-studentized_range-arg307-None-None] - AttributeError: 'NoneType' object has no attribute 'lower'
FAILED scipy/stats/tests/test_continuous_basic.py::test_cont_basic_fit[200-genhyperbolic-arg108-None-None] - AttributeError: 'NoneType' object has no attribute 'lower'
FAILED scipy/stats/tests/test_continuous_basic.py::test_cont_basic_fit[200-gausshyper-arg94-None-None] - AttributeError: 'NoneType' object has no attribute 'lower'
FAILED scipy/stats/tests/test_continuous_basic.py::test_cont_basic_fit[200-genexpon-arg95-None-None] - AttributeError: 'NoneType' object has no attribute 'lower'
FAILED scipy/stats/tests/test_continuous_basic.py::test_cont_basic_fit[200-exponweib-arg72-MM-None] - scipy.integrate._quadpack_py.IntegrationWarning: The integral is probably d...
FAILED scipy/stats/tests/test_continuous_basic.py::test_cont_basic_fit[200-vonmises_line-arg340-None-None] - AttributeError: 'NoneType' object has no attribute 'lower'
FAILED scipy/stats/tests/test_continuous_basic.py::test_cont_basic_fit[200-johnsonsu-arg175-MM-None] - scipy.integrate._quadpack_py.IntegrationWarning: The algorithm does not con...
FAILED scipy/stats/tests/test_continuous_basic.py::test_cont_basic_fit[200-kappa4-arg176-None-None] - AttributeError: 'NoneType' object has no attribute 'lower'
FAILED scipy/stats/tests/test_continuous_basic.py::test_cont_basic_fit[200-kappa4-arg177-None-None] - AttributeError: 'NoneType' object has no attribute 'lower'
FAILED scipy/stats/tests/test_continuous_basic.py::test_cont_basic_fit[200-kappa4-arg178-None-None] - AttributeError: 'NoneType' object has no attribute 'lower'
============ 19 failed, 323 passed, 30 skipped in 326.13s (0:05:26) ============

```

</details>


### SciPy/NumPy/Python version and system information

```shell
1.14.0.dev0+1021.bd60b4e 1.26.4 sys.version_info(major=3, minor=12, micro=3, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: pkgconfig
    found: true
    include directory: /usr/include/x86_64-linux-gnu/openblas-pthread/
    lib directory: /usr/lib/x86_64-linux-gnu/openblas-pthread/
    name: openblas
    openblas configuration: USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER=1 NO_CBLAS=
      NO_LAPACK= NO_LAPACKE=1 NO_AFFINITY=1 USE_OPENMP=0 generic MAX_THREADS=64
    pc file directory: /usr/lib/x86_64-linux-gnu/pkgconfig
    version: 0.3.20
  lapack:
    detection method: pkgconfig
    found: true
    include directory: /usr/include/x86_64-linux-gnu/openblas-pthread/
    lib directory: /usr/lib/x86_64-linux-gnu/openblas-pthread/
    name: openblas
    openblas configuration: USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER=1 NO_CBLAS=
      NO_LAPACK= NO_LAPACKE=1 NO_AFFINITY=1 USE_OPENMP=0 generic MAX_THREADS=64
    pc file directory: /usr/lib/x86_64-linux-gnu/pkgconfig
    version: 0.3.20
  pybind11:
    detection method: config-tool
    include directory: unknown
    name: pybind11
    version: 2.12.0
Compilers:
  c:
    commands: cc
    linker: ld.bfd
    name: gcc
    version: 11.4.0
  c++:
    commands: c++
    linker: ld.bfd
    name: gcc
    version: 11.4.0
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 3.0.10
  fortran:
    commands: gfortran
    linker: ld.bfd
    name: gcc
    version: 11.4.0
  pythran:
    include directory: ../../../../../py3.12.3/lib/python3.12/site-packages/pythran
    version: 0.15.0
Machine Information:
  build:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
  cross-compiled: false
  host:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
Python Information:
  path: /home/warren/py3.12.3/bin/python3.12
  version: '3.12'
```

--- a/scipy/stats/tests/test_continuous_basic.py
+++ b/scipy/stats/tests/test_continuous_basic.py
@@ -51,21 +51,22 @@ slow_test_cont_basic = {'crystalball', 'powerlognorm', 'pearson3'}
 xslow_test_moments = {'studentized_range', 'ksone', 'vonmises', 'vonmises_line',
                       'recipinvgauss', 'kstwo', 'kappa4'}
 
-xslow_fit_all = {'ksone', 'kstwo', 'levy_stable', 'recipinvgauss', 'studentized_range',
-                 'gausshyper', 'kappa4', 'vonmises_line', 'genhyperbolic', 'ncx2',
-                 'powerlognorm', 'genexpon'}
-xfail_fit_all = {'trapezoid', 'truncpareto'}
-xslow_fit_mm = {'argus', 'beta', 'exponpow', 'exponweib', 'gengamma',
+xslow_fit_mle = {'gausshyper', 'ncf', 'ncx2', 'vonmises_line'}
+xfail_fit_mle = {'ksone', 'kstwo', 'trapezoid', 'truncpareto'}
+skip_fit_mle = {'levy_stable', 'studentized_range'}  # far too slow (>10min)
+xslow_fit_mm = {'argus', 'beta', 'exponpow', 'gausshyper', 'gengamma',
                 'genhalflogistic', 'geninvgauss', 'gompertz', 'halfgennorm',
-                'johnsonsb', 'johnsonsu', 'powernorm', 'vonmises',
-                'truncnorm', 'kstwobign', 'rel_breitwigner', 'wrapcauchy',
-                'johnsonsu', 'truncweibull_min', 'truncexpon', 'norminvgauss'}
-xslow_fit_mle = {'ncf'}
-xfail_fit_mm = {'alpha', 'betaprime', 'burr', 'burr12', 'cauchy', 'crystalball', 'f',
-                'fisk', 'foldcauchy', 'genextreme', 'genpareto', 'halfcauchy',
-                'invgamma', 'jf_skew_t', 'kappa3', 'levy', 'levy_l', 'loglaplace',
-                'lomax', 'mielke', 'ncf', 'nct', 'pareto', 'skewcauchy', 't',
-                'bradford', 'tukeylambda'}
+                'johnsonsb', 'kstwobign', 'ncx2', 'norminvgauss', 'truncnorm',
+                'truncweibull_min', 'wrapcauchy'}
+xfail_fit_mm = {'alpha', 'betaprime', 'bradford', 'burr', 'burr12', 'cauchy',
+                'crystalball', 'exponweib', 'f', 'fisk', 'foldcauchy', 'genextreme',
+                'genpareto', 'halfcauchy', 'invgamma', 'jf_skew_t', 'johnsonsu',
+                'kappa3', 'kappa4', 'levy', 'levy_l', 'loglaplace', 'lomax', 'mielke',
+                'ncf', 'nct', 'pareto', 'powerlognorm', 'powernorm', 'rel_breitwigner',
+                'skewcauchy', 't', 'trapezoid', 'truncexpon', 'truncpareto',
+                'tukeylambda', 'vonmises', 'vonmises_line'}
+skip_fit_mm = {'genexpon', 'genhyperbolic', 'ksone', 'kstwo', 'levy_stable',
+               'recipinvgauss', 'studentized_range'}  # far too slow (>10min)
 
 # These distributions fail the complex derivative test below.
 # Here 'fail' mean produce wrong results and/or raise exceptions, depending
@@ -199,30 +200,42 @@ def test_cont_basic(distname, arg, sn):
 
 
 def cases_test_cont_basic_fit():
-    message = "Test fails and may be slow"
-    fail_mark = pytest.mark.skip(reason=message)
+    xslow = pytest.mark.xslow
+    fail = pytest.mark.skip(reason="Test fails and may be slow.")
+    skip = pytest.mark.skip(reason="Test too slow to run to completion (>10m).")
 
     for distname, arg in distcont[:] + histogram_test_instances:
-        if distname in xfail_fit_all:
-            yield pytest.param(distname, arg, None, None, marks=fail_mark)
-            continue
-        if distname in xslow_fit_all:
-            yield pytest.param(distname, arg, None, None, marks=pytest.mark.xslow)
-            continue
-
         for method in ["MLE", "MM"]:
-            if method == 'MM' and distname in xfail_fit_mm:
-                yield pytest.param(distname, arg, method, None, marks=fail_mark)
-                continue
-            if method == 'MM' and distname in xslow_fit_mm:
-                yield pytest.param(distname, arg, method, None, marks=pytest.mark.xslow)
-                continue
-            if method == 'MLE' and distname in xslow_fit_mle:
-                yield pytest.param(distname, arg, method, None, marks=pytest.mark.xslow)
-                continue
-
             for fix_args in [True, False]:
+                if method == 'MLE' and distname in xslow_fit_mle:
+                    yield pytest.param(distname, arg, method, fix_args, marks=xslow)
+                    continue
+                if method == 'MLE' and distname in xfail_fit_mle:
+                    yield pytest.param(distname, arg, method, fix_args, marks=fail)
+                    continue
+                if method == 'MLE' and distname in skip_fit_mle:
+                    yield pytest.param(distname, arg, method, fix_args, marks=skip)
+                    continue
+                if method == 'MM' and distname in xslow_fit_mm:
+                    yield pytest.param(distname, arg, method, fix_args, marks=xslow)
+                    continue
+                if method == 'MM' and distname in xfail_fit_mm:
+                    yield pytest.param(distname, arg, method, fix_args, marks=fail)
+                    continue
+                if method == 'MM' and distname in skip_fit_mm:
+                    yield pytest.param(distname, arg, method, fix_args, marks=skip)
+                    continue
+
                 yield distname, arg, method, fix_args
+
+
+def test_cont_basic_fit_cases():
+    # Distribution names should not be in multiple MLE or MM sets
+    assert (len(xslow_fit_mle.union(xfail_fit_mle).union(skip_fit_mle)) ==
+            len(xslow_fit_mle) + len(xfail_fit_mle) + len(skip_fit_mle))
+    assert (len(xslow_fit_mm.union(xfail_fit_mm).union(skip_fit_mm)) ==
+            len(xslow_fit_mm) + len(xfail_fit_mm) + len(skip_fit_mm))
+
 
 @pytest.mark.parametrize('distname, arg, method, fix_args',
                          cases_test_cont_basic_fit())

